core/tx_pool.go

```diff
@@ -1427,6 +1428,27 @@ func (pool *TxPool) promoteExecutables(accounts []common.Address) []*types.Trans
 	return promoted
 }
 
+func (pool *TxPool) executableTxFilter(addr common.Address) func(tx *types.Transaction) bool {
+	costLimit := pool.currentState.GetBalance(addr)
```

Comment:
Here maybe cause this case:

The caller logic 
```
f := pool.executableTxFilter(....)
// do some check, maybe need 20ms, like
Sleep(20ms)
// Then
f()
```
When f() begin to execute, the `costLimit` before  20ms .

Why don't put `costLimit := pool.currentState.GetBalance(addr)` to `func(tx *types.Transaction) bool {`

Reply:
thats the right place to put it i guess

Reply:
For current logic of scroll-tech/go-ethereum, put here is no issue. If caller call it like the above example, I think it's not right.